x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlAssignmentNotifier.java

```diff
@@ -25,57 +24,39 @@
 import org.elasticsearch.xpack.ml.notifications.Auditor;
 
 import java.util.Objects;
-import java.util.concurrent.atomic.AtomicBoolean;
 
-public class MlAssignmentNotifier implements ClusterStateListener, LocalNodeMasterListener {
+public class MlAssignmentNotifier implements ClusterStateListener {
     private static final Logger logger = LogManager.getLogger(MlAssignmentNotifier.class);
 
     private final Auditor auditor;
-    private final ClusterService clusterService;
     private final MlConfigMigrator mlConfigMigrator;
     private final ThreadPool threadPool;
-    private final AtomicBoolean enabled = new AtomicBoolean(false);
 
     MlAssignmentNotifier(Auditor auditor, ThreadPool threadPool, Client client, ClusterService clusterService) {
         this.auditor = auditor;
-        this.clusterService = clusterService;
         this.mlConfigMigrator = new MlConfigMigrator(client, clusterService);
         this.threadPool = threadPool;
-        clusterService.addLocalNodeMasterListener(this);
+        clusterService.addListener(this);
     }
 
     MlAssignmentNotifier(Auditor auditor, ThreadPool threadPool, MlConfigMigrator mlConfigMigrator, ClusterService clusterService) {
         this.auditor = auditor;
-        this.clusterService = clusterService;
         this.mlConfigMigrator = mlConfigMigrator;
         this.threadPool = threadPool;
-        clusterService.addLocalNodeMasterListener(this);
+        clusterService.addListener(this);
     }
 
-    @Override
-    public void onMaster() {
```

Comment:
The rolling upgrade tests occasionally fail because the migration does not get triggered. I suspect the cause could have been that the `onMaster` method of  `LocalNodeMasterListener` which enables migration is called after the vital cluster state change event caused by master election. I changed this so it is no longer a `LocalNodeMasterListener` instead it ignores events if it is not the master.